src/arena.c

```diff
@@ -1708,7 +1734,8 @@ arena_dalloc_bin_locked_impl(tsdn_t *tsdn, arena_t *arena, extent_t *slab,
 
 	arena_slab_reg_dalloc(tsdn, slab, slab_data, ptr);
 	unsigned nfree = extent_nfree_get(slab);
-	if (nfree == bin_info->nregs) {
+	if (nfree == bin_info->nregs && !sized_alloc_region_lookup(
```

Comment:
Yeah. All this logic got pushed down into helper functions, which means we don't care about slab location here. Fixed.